clients/src/test/java/org/apache/kafka/clients/admin/AdminClientUnitTestEnv.java

```diff
@@ -113,6 +114,9 @@ public MockClient kafkaClient() {
 
     @Override
     public void close() {
+        // tell the admin client to close now
+        this.adminClient.close(Duration.ZERO);
+        // block until it is really closed
         this.adminClient.close();
```

Comment:
Seems like we should do a single `close` with some timeout. cc @cmccabe

Reply:
I considered it, but at the same time, it seemed nice to actually wait until the admin client's thread exits, so we don't get too many threads stacked up.

Reply:
What if that never happens? It's better to always have a timeout in tests.

Reply:
Fair enough. I'll change the second close to time out after the default timeout of 2 minutes.